features/karaf/esh-tp/src/main/feature/feature.xml

```diff
@@ -141,11 +148,11 @@
   </feature>
 
   <feature name="esh-tp-jupnp" description="UPnP/DLNA library for Java" version="${project.version}">
-    <capability>esh.tp;feature=jupnp;version=2.2.0</capability>
-
+    <capability>esh.tp;feature=jupnp;version=2.3.0</capability>
     <feature dependency="true">http</feature>
     <feature dependency="true">esh-tp-httpclient</feature>
-    <bundle>mvn:org.jupnp/org.jupnp/2.2.0</bundle>
+    <bundle dependency="true">mvn:org.osgi/org.osgi.compendium/4.2.0</bundle>
```

Comment:
@maggu2810 Do you have any advice how to correctly handle this? org.jupnp has a dependency on `org.osgi.component`. How do I express this in a neutral way? I somehow couldn't figure out how all our other OSGi dependencies are resolved, we seem to refer to the compendium spec nowhere in the feature definitions...

Reply:
You should never depend on an API bundle.
As we don't want to force a specific implementation we already use `dependency="true"`, so the referenced dependency it is only installed if no other one satisfy the requirements.
For a the Component / DS part, you should use the SCR feature.

Does this one work for you:
```text
<feature dependency="true">scr</feature>
```

Reply:
Thank you, seems to work! I have updated the PR accordingly.